File: app/src/main/java/io/ipoli/android/store/Avatar.java
package io.ipoli.android.store;

import android.support.annotation.DrawableRes;
import android.support.annotation.StringRes;

import io.ipoli.android.R;

/**
 * Created by Polina Zhelyazkova <polina@ipoli.io>
 * on 5/25/17.
 */

public enum Avatar {
    IPOLI_CLASSIC(1, 0, R.string.pet_donkey, R.drawable.avatar_12),
    IPOLI_YELLOW_GLASSES(2, 0, R.string.pet_donkey, R.drawable.avatar_11),
    MACARENA(3, 20, R.string.pet_donkey, R.drawable.avatar_10),
    MACARENA_MEXICAN(4, 20, R.string.pet_donkey, R.drawable.avatar_09),
    BLONDY(5, 20, R.string.pet_donkey, R.drawable.avatar_08),
    GREEN_EYES(6, 20, R.string.pet_donkey, R.drawable.avatar_07),
    PIPILOTA(7, 20, R.string.pet_donkey, R.drawable.avatar_06),
    OLD_PIRATE(8, 20, R.string.pet_donkey, R.drawable.avatar_05),
    BEARD_GUY(9, 20, R.string.pet_donkey, R.drawable.avatar_04),
    DWIGHT(10, 20, R.string.pet_donkey, R.drawable.avatar_03),
    MICHAEL(11, 20, R.string.pet_donkey, R.drawable.avatar_02),
    TOBBY(12, 20, R.string.pet_donkey, R.drawable.avatar_01);

    public final int code;
    public final int price;
    public final int name;
    public final int picture;

    Avatar(int code, int price, @StringRes int name, @DrawableRes int picture) {
        this.code = code;
        this.price = price;
        this.name = name;
        this.picture = picture;
    }
}

File: app/src/main/java/mypoli/android/player/data/Avatar.kt
package mypoli.android.player.data

import android.support.annotation.DrawableRes
import android.support.annotation.StringRes
import mypoli.android.R

/**
 * Created by Polina Zhelyazkova <polina@mypoli.fun>
 * on 8/21/17.
 */
enum class Avatar(val gemPrice: Int) {
    IPOLI_CLASSIC(0),
    IPOLI_YELLOW_GLASSES(0),
    MACARENA(8),
    MACARENA_MEXICAN(8),
    BLONDY(8),
    GREEN_EYES(8),
    PIPILOTA(8),
    OLD_PIRATE(8),
    BEARDY(8),
    DWIGHT(8),
    MICHAEL(8),
    TOBBY(8);
}

enum class AndroidAvatar(
    @StringRes val avatarName: Int,
    @DrawableRes val image: Int
) {
    IPOLI_CLASSIC(R.string.avatar_name_ipoli_classic, R.drawable.avatar_12),
    IPOLI_YELLOW_GLASSES(R.string.avatar_name_ipoli_yellow_glasses, R.drawable.avatar_11),
    MACARENA(R.string.avatar_name_macarena, R.drawable.avatar_10),
    MACARENA_MEXICAN(R.string.avatar_name_macarena_mexican, R.drawable.avatar_09),
    BLONDY(R.string.avatar_name_blondy, R.drawable.avatar_08),
    GREEN_EYES(R.string.avatar_name_green_eyes, R.drawable.avatar_07),
    PIPILOTA(R.string.avatar_name_pipilota, R.drawable.avatar_06),
    OLD_PIRATE(R.string.avatar_name_old_pirate, R.drawable.avatar_05),
    BEARDY(R.string.avatar_beardy, R.drawable.avatar_04),
    DWIGHT(R.string.avatar_name_dwight, R.drawable.avatar_03),
    MICHAEL(R.string.avatar_name_michael, R.drawable.avatar_02),
    TOBBY(R.string.avatar_name_tobby, R.drawable.avatar_01);
}